modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -75,13 +74,21 @@ public void validate(ExtensionModel model) throws IllegalModelDefinitionExceptio
 
   private class ValidationDelegate {
 
+    public static final String SINGULARIZED_CLASH_MESSAGE =
+        "Extension '%s' contains %d parameters that clash when singularized. %s";
+    public static final String NAME_CLASH_MESSAGE =
+        "%s '%s' contains parameter '%s' that when transformed into DSL language clashes with parameter '%s' from '%s'";
+
     private final ExtensionModel extensionModel;
     private final Set<DescribedReference<NamedObject>> namedObjects = new HashSet<>();
     private final Map<String, DescribedParameter> singularizedObjects = new HashMap<>();
     private final Multimap<String, TopLevelParameter> topLevelParameters = LinkedListMultimap.create();
+    private final DslSyntaxResolver dslSyntaxResolver;
 
     public ValidationDelegate(ExtensionModel extensionModel) {
       this.extensionModel = extensionModel;
+      this.dslSyntaxResolver = new DslSyntaxResolver(extensionModel,
+                                                     new SingleExtensionImportTypesStrategy());
```

Comment:
The second parameter is how you configure the DslSyntaxResolver to work without any imported type without failing. Check if that's the way to go.